test/index.js

```diff
@@ -38,80 +31,114 @@ describe('FileLockedOperation', function() {
+      };
 
-        var checkErrorMsg = check(restorePermissions, function(err) {
-          expect(err.message).to.equal(
-            'FileLockedOperation.doLockedOperation: EACCES, open \'' +
-            lockFilePath + '\'');
+      restorePermissions = function(operationError) {
+        return new Promise(function(resolve, reject) {
+          fs.chmod(lockFileDir, '700', function(err) {
+            if (err) {
+              return reject(err);
+            }
+            operationError ? reject(operationError) : resolve();
+          });
         });
+      };
 
-        lock.doLockedOperation(function() { }, checkErrorMsg);
-      });
+      return changePermissions()
+        .then(function() {
+          return lock.doLockedOperation(function() { });
+        })
+        .then(restorePermissions, restorePermissions)
+        .should.be.rejectedWith(Error, lockFilePath);
     });
 
     it('should prevent multiple operations from overlapping', function(done) {
       // We generate Promise objects in order to create a series of
       // interleaved asynchronous operations. The trick is to wrap the
       // generator() calls in function literals passed to Promise.then().
-      var ops = [];
-      var generator = function(op) {
-        return new Promise(function(resolve) { ops.push(op); resolve(); });
+      var ops = [],
+          generator,
+          combinedOps;
+
+      generator = function(op) {
+        return new Promise(function(resolve) {
+          ops.push(op);
+          resolve();
+        });
       };
 
-      var combinedOps = function(done) {
+      combinedOps = function() {
         // To make this test fail (because the operations are interleaved),
-        // then delete/comment out the lock.doLockedOperation() call.
-        lock.doLockedOperation(function(done) {
+        // delete/comment out the lock.doLockedOperation() call.
+        return lock.doLockedOperation(function() {
           generator('op1')
             .then(function() { return generator('op2'); })
-            .then(function() { return generator('op3'); })
-            .then(done, done);
-        }, done);
+            .then(function() { return generator('op3'); });
+        });
       };
 
-      // checkCallsDoNotOverlap will execute the assertions after all of the
-      // Promises have been resolved.
-      var checkCallsDoNotOverlap = checkN(3, done, function(errors) {
-        expect(errors).to.be.empty;
-        expect(ops.slice(0, 3)).to.eql(ops.slice(3, 6));
-        expect(ops.slice(3, 6)).to.eql(ops.slice(6, 9));
-      });
-
-      combinedOps(checkCallsDoNotOverlap);
-      combinedOps(checkCallsDoNotOverlap);
-      combinedOps(checkCallsDoNotOverlap);
+      Promise.all([combinedOps(), combinedOps(), combinedOps()])
+        .should.be.fulfilled.then(function() {
+          ops.slice(0, 3).should.eql(ops.slice(3, 6));
+          ops.slice(3, 6).should.eql(ops.slice(6, 9));
+        })
+        .should.notify(done);
```

Comment:
FYI that Mocha has an even simpler construct for working with Promises:

https://mochajs.org/#working-with-promises

i.e. just return the Promise rather than needing to call `done`.